pkg/kubelet/kubelet.go

```diff
@@ -110,8 +110,8 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
-	// nodeStatusUpdateRetry specifies how many times kubelet retries when posting node status failed.
-	nodeStatusUpdateRetry = 5
+	// nodeStatusUpdateRetryBackoff is the amount of time that we are willing to spend retrying to connect to the apiserver.
+	nodeStatusUpdateRetryMaximumTime = 20 * time.Minute
```

Comment:
> Is that necessarily a bad thing?

In my opinion it is a bad thing. Basically you may have a pod that doesn't need network at all to run (e.g. some batch job). And we don't want to evict it.